src/app/shared/components/status-info-card/status-info-card.component.ts

```diff
@@ -3,17 +3,24 @@ import { ApplicationIcons } from '../../enum/applications';
 import { ApplicationStatusDescription, ApplicationTitles } from '../../enum/enumUA/statuses';
 import { ApplicationStatuses } from '../../enum/statuses';
 import { Application } from '../../models/application.model';
+import { Role } from '../../enum/role';
 
 @Component({
   selector: 'app-status-info-card',
   templateUrl: './status-info-card.component.html',
   styleUrls: ['./status-info-card.component.scss']
 })
 export class StatusInfoCardComponent {
+  @Input() public application: Application = null;
+  @Input() public role: Role;
+
   public readonly statusTitles = ApplicationTitles;
   public readonly applicationStatusDescription = ApplicationStatusDescription;
   public readonly applicationIcons = ApplicationIcons;
-  public readonly statuses = ApplicationStatuses;
-
-  @Input() application: Application = null;
+  public readonly Role = Role;
+  public get statuses(): any {
```

Comment:
This is not exactly what I thinking.
In the example below, we will update the statuses only when the role is set.
```
private _role: Role;
  public get role(): Role {
    return this._role;
  }
  @Input()
  public set role(value: Role) {
    this._role = value;
    this.setStatuses(); // can be same code, as in lines 22-24
  }
```